devito/data/decomposition.py

```diff
@@ -241,6 +249,24 @@ def convert_index(self, *args, rel=True):
                 if glb_idx_max is not None and glb_idx_max < 0:
                     glb_idx_max = self.glb_max + glb_idx_max + 1
                 # -> Do the actual conversion
+                # First check for the special case of a slice with negative step
+                if isinstance(glb_idx, slice) and glb_idx.step < 0:
+                    if glb_idx_max is None or glb_idx_max > self.loc_abs_max:
```

Comment:
Now that I look at this more carefully...are these lines, 254~259 (extremes included), identical to lines 276~281?

Reply:
Indeed. I've tidied the code a bit now.